ApiFirst/ApiFirst/RepositoryList.cs

```diff
@@ -0,0 +1,180 @@
+﻿using DomainModels.DataBase;
+using DomainModels.Models;
+using DomainServices.Interfaces;
+
+namespace DomainServices.Repositories
+{
+    public class RepositoryList : ICustomerRepository
+    {
+        public int Create(Customer model)
+        {
+            model.Cpf = model.Cpf.Trim().Replace(".", "").Replace("-", "");
+            int response = 409;
+
+            ListCustomer list = ListCustomer.GetInstance();
+
+            model.Id = list.GetCustomerList().Count;
+
+
+            if (!list.GetCustomerList().Any())
+            {
+                model.Id = 0;
+                GetAll().Add(model);
+                return 201;
+            }
+
+            foreach (Customer item in list.GetCustomerList())
+            {
+                if (model.Cpf != item.Cpf && model.Email != item.Email)
+                {
+                    GetAll().Add(model);
+                    response = 201;
+                }
+                return response;
+            }
+
+            return 400;
+        }
+
+        public int DeleteFromList(string cpf, string email)
+        {
+            cpf = cpf.Trim().Replace(".", "").Replace("-", "");
+
+            if (!GetAll().Any())
+            {
+                return 400;
+            }
+
+            foreach (Customer item in GetAll())
+            {
+                if (cpf == item.Cpf && email == item.Email)
+                {
+                    GetAll().Remove(item);
+                    return 200;
+                }
+            }
+            return 404;
+        }
+        public int Delete(int id)
+        {
+
+            if (!GetAll().Any())
+            {
+                return 400;
+            }
+
+            if (GetAll().Count() >= id)
+            {
+                GetAll().Remove(GetAll().ElementAt(id));
+                return 200;
+            }
+
+            return 404;
```

Comment:
Aqui só há dois retornos: Ok ou NotFound. É melhorar a exclusão com a função abaixo:
```suggestion
            var index = list.FindIndex(customer => customer.Id == id);
            if(index == -1) return 404;
            list.RemoveAt(index);
            return 200;
```